code/modules/mod/modules/modules_science.dm

```diff
@@ -159,3 +159,167 @@
+			var/atom/core_path = path
+			if(path == accepted_anomalies[1])
+				core_string += initial(core_path.name)
+			else if(path == accepted_anomalies[length(accepted_anomalies)])
+				core_string += "or [initial(core_path.name)]"
+			else
+				core_string += ", [initial(core_path.name)]"
+		. += span_notice("You need to insert \a [core_string] for this module to function.")
+
+/obj/item/mod/module/anomaly_locked/on_select()
+	if(!core)
+		balloon_alert(mod.wearer, "no core!")
+		return
+	return ..()
+
+/obj/item/mod/module/anomaly_locked/on_process(delta_time)
+	. = ..()
+	if(!core)
+		return FALSE
+
+/obj/item/mod/module/anomaly_locked/on_active_process(delta_time)
+	if(!core)
+		return FALSE
+	return TRUE
+
+/obj/item/mod/module/anomaly_locked/attackby(obj/item/item, mob/living/user, params)
+	if(item.type in accepted_anomalies)
+		if(core)
+			balloon_alert(user, "core already in!")
+			return
+		if(!user.transferItemToLoc(item, src))
+			return
+		core = item
+		balloon_alert(user, "core installed")
+		playsound(src, 'sound/machines/click.ogg', 30, TRUE)
+		update_icon_state()
+	else
+		return ..()
+
+/obj/item/mod/module/anomaly_locked/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(!core)
+		balloon_alert(user, "no core!")
+		return
+	balloon_alert(user, "removing core...")
+	if(!do_after(user, 3 SECONDS, target = src))
+		balloon_alert(user, "interrupted!")
+		return
+	balloon_alert(user, "core removed")
+	core.forceMove(drop_location())
+	if(Adjacent(user) && !issilicon(user))
+		user.put_in_hands(core)
+	core = null
+	update_icon_state()
+
+/obj/item/mod/module/anomaly_locked/update_icon_state()
+	icon_state = initial(icon_state) + (core ? "-core" : "")
+	return ..()
+
+///Anti-Gravity - Makes the user weightless.
+/obj/item/mod/module/anomaly_locked/antigrav
+	name = "MOD anti-gravity module"
+	desc = "A module that uses a gravitational core to make the user completely weightless."
+	icon_state = "antigrav"
+	module_type = MODULE_TOGGLE
+	complexity = 3
+	active_power_cost = DEFAULT_CELL_DRAIN * 0.7
+	cooldown_time = 0.5 SECONDS
+	accepted_anomalies = list(/obj/item/assembly/signaler/anomaly/grav)
+
+/obj/item/mod/module/anomaly_locked/antigrav/on_activation()
+	. = ..()
+	if(!.)
+		return
+	if(mod.wearer.has_gravity())
+		new /obj/effect/temp_visual/mook_dust(get_turf(src))
+	mod.wearer.AddElement(/datum/element/forced_gravity, 0)
+	mod.wearer.update_gravity(mod.wearer.has_gravity())
+	playsound(src, 'sound/effects/gravhit.ogg', 50)
+
+/obj/item/mod/module/anomaly_locked/antigrav/on_deactivation()
+	. = ..()
+	if(!.)
+		return
+	mod.wearer.RemoveElement(/datum/element/forced_gravity, 0)
+	mod.wearer.update_gravity(mod.wearer.has_gravity())
+	if(mod.wearer.has_gravity())
+		new /obj/effect/temp_visual/mook_dust(get_turf(src))
+	playsound(src, 'sound/effects/gravhit.ogg', 50)
+
+/obj/item/mod/module/anomaly_locked/antigrav/prebuilt
+	prebuilt = TRUE
+
+#define TELEPORT_TIME 3 SECONDS
```

Comment:
bonus idea: change this from a define to a var on the MOD, so admins can bus it / future subtypes can make speedy teleporters